src/components/shared/CustomImage.tsx

```diff
@@ -0,0 +1,52 @@
+import { Box } from '@mui/material';
+import { useEffect, useState } from 'react';
+import LoadingImage from '../../assets/loader.png';
+import ErrorImage from '../../assets/no-image.png';
+
+const imageStates = {
+  Loading: LoadingImage,
+  Error: ErrorImage,
+  Loaded: '',
+};
+
+type ImageState = keyof typeof imageStates;
+type SingleImageProps = { image: string; name?: string };
+
+function CustomImage({ image, name = 'image' }: SingleImageProps) {
+  const [currentState, setCurrentState] = useState<ImageState>('Loading');
```

Comment:
Looks like `currentState` will have a `Loading` state only once during the initial render. 

When image prop will be changed, there will be no `Loading` value. 

Is it done by purpose or you missed that?